### PR TITLE
[RORDEV-76] Added info about LDAP server discovery feature

### PR DESCRIPTION
**Pull request by pielas**

### CODE REVIEW

**Review by coutoPL — APPROVED** [review]

**Comment by coutoPL** [comment]
@sscarduzio do you have any comments?

File: elasticsearch.md [hunk 1]
```diff
@@ -1759,6 +1765,37 @@ group_search_filter: "(cn=*)" # basically no group filtering
 
 Caching can be configured per LDAP client \(see `ldap1`\) or per rule \(see `Accept requests from users in group team2 on index2` rule\)
 
+#### LDAP Server discovery
+
+It is possible for LDAP module to get all LDAP hostnames from DNS server rather than from configuration file. By default `_ldap._tcp` SRV records are used for that, but any other SRV record can be configured.
```
Comment (sscarduzio) [inline elasticsearch.md:1770]: the LDAP connector

File: elasticsearch.md [hunk 2]
```diff
@@ -1759,6 +1765,37 @@ group_search_filter: "(cn=*)" # basically no group filtering
...
+
+The simplest configuration of LDAP using server discovery is:
```
Comment (sscarduzio) [inline elasticsearch.md:1772]: The simplest configuration example of an LDAP connector instance using...

File: elasticsearch.md [hunk 3]
```diff
@@ -1759,6 +1765,37 @@ group_search_filter: "(cn=*)" # basically no group filtering
...
+```
+    - name: ldap
+      server_discovery: true                                        
+      search_user_base_DN: "ou=People,dc=example2,dc=com"
+      search_groups_base_DN: "ou=Groups,dc=example2,dc=com"
+```
+It is using system DNS to get `_ldap._tcp` SRV records which should contain hostnames and ports to connect to.
```
Comment (sscarduzio) [inline elasticsearch.md:1779]: This configuration is using the system DNS to fetch all the...

Comment (sscarduzio) [inline elasticsearch.md:1779]: ... which are expected to contain the hostname and port of all the LDAP servers we should connect to.

File: elasticsearch.md [hunk 4]
```diff
@@ -1759,6 +1765,37 @@ group_search_filter: "(cn=*)" # basically no group filtering
...
+
```
Comment (sscarduzio) [inline elasticsearch.md:1780]: Maybe talk about the endpoint rotation policy: round robin or failover, etc.

File: elasticsearch.md [hunk 5]
```diff
@@ -1759,6 +1765,37 @@ group_search_filter: "(cn=*)" # basically no group filtering
...
+Server discovery could be personalised using few **optional** configuration parameters:
```
Comment (sscarduzio) [inline elasticsearch.md:1781]: The server discovery mechanism can be optionally configured further, by adding a few more configuration parameters:

File: elasticsearch.md [hunk 6]
```diff
@@ -1759,6 +1765,37 @@ group_search_filter: "(cn=*)" # basically no group filtering
...
+* `record_name` - DNS SRV record name. By default it's `_ldap._tcp`, but could be `_ldap._tcp.domainname` or any custom value.
+* `dns_url` - Address of non-default DNS server in form `dns://IP[:PORT]`. By default system DNS is used.
```
Comment (sscarduzio) [inline elasticsearch.md:1783]: the system DNS....

**Review by sscarduzio — COMMENTED** [review]
Thank you!

File: elasticsearch.md (hunk 5 above)
Comment (pielas) [reply elasticsearch.md:1781]: ok, I also added ", all of which are optional" to emphasize their optionality

**Pull request merged**